woocommerce/woocommerce-gutenberg-products-block
# Update atomic block structure

_#2399 is a mammoth PR so I'm breaking out some fixes and changes to review separately._

This PR moves some code around and restructures the Atomic Blocks. Why? Because in order to render Inner Blocks without a LayoutConfig, we need some extra data shared between the component and the block (mainly attributes). The PR contains the following changes:

- Moves components from atomic/components/<component>/index to atomic/blocks/<block>/block.js. This also more closely matches other non-atomic blocks
- Adds an alias for `@woocommerce/atomic-blocks` which is used to import the blocks themselves.
- Adds a map of block components so we can map block->component when rendering layouts still
- Splits the atomic block files up into edit.js, attributes.js, from the existing index.js. The code is unchanged, just moved.

I have not yet added context to this components— that will be done in a follow up.

### How to test the changes in this Pull Request:

Since this changes structure only, testing should be focussed around All Products block. Ensure existing blocks continue to work (edit the layout), and render correctly on the frontend.
